api/circulation.py

```diff
@@ -202,7 +202,7 @@ def borrow(self, patron, pin, licensepool, delivery_mechanism,
         if patron.fines:
             def parse_fines(fines):
                 dollars, cents = re.match("\$([\d]+)\.(\d\d)", fines).groups()
-                return (dollars * 100) + cents
+                return (int(dollars) * 100) + int(cents)
```

Comment:
This was working most of the time when two amounts of fines were compared, but it worked by multiplying a string like "20" 100 times, "2020202020202020202020...", not by actually comparing two amounts of money as numeric values.